otelcollector/configmapparser/tomlparser-scrape-interval.rb

```diff
@@ -0,0 +1,166 @@
+#!/usr/local/bin/ruby
+# frozen_string_literal: true
+
+require "tomlrb"
+if (!ENV["OS_TYPE"].nil? && ENV["OS_TYPE"].downcase == "linux")
+  require "re2"
+end
+require "yaml"
+require_relative "ConfigParseErrorLogger"
+
+LOGGING_PREFIX = "default-scrape-interval-list"
+
+@configMapMountPath = "/etc/config/settings/default-targets-scrape-interval-settings"
+@configVersion = ""
+@configSchemaVersion = ""
+
+@kubeletScrapeInterval = "30s"
+@corednsScrapeInterval = "30s"
+@cadvisorScrapeInterval = "30s"
+@kubeproxyScrapeInterval = "30s"
+@apiserverScrapeInterval = "30s"
+@kubestateScrapeInterval = "30s"
+@nodeexporterScrapeInterval = "30s"
+@windowsexporterScrapeInterval = "30s"
+@windowskubeproxyScrapeInterval = "30s"
+@prometheusCollectorHealthInterval = "30s"
+
+# Use parser to parse the configmap toml file to a ruby structure
+def parseConfigMap
+  begin
+    # Check to see if config map is created
+    if (File.file?(@configMapMountPath))
+      parsedConfig = Tomlrb.load_file(@configMapMountPath, symbolize_keys: true)
+      return parsedConfig
+    else
+      ConfigParseErrorLogger.log(LOGGING_PREFIX, "configmap prometheus-collector-configmap for default-targets-scrape-interval-settings not mounted, using defaults")
+      return nil
+    end
+  rescue => errorStr
+    ConfigParseErrorLogger.logError(LOGGING_PREFIX, "Exception while parsing config map for default-targets-scrape-interval-settings: #{errorStr}, using defaults, please check config map for errors")
+    return nil
+  end
+end
+
+# Use the ruby structure created after config parsing to set the right values to be used for otel collector settings
+def populateSettingValuesFromConfigMap(parsedConfig)
+  begin
+    kubeletScrapeInterval = parsedConfig[:kubelet]
+    if !kubeletScrapeInterval.nil?
+        @kubeletScrapeInterval = kubeletScrapeInterval
+        ConfigParseErrorLogger.log(LOGGING_PREFIX, "Using configmap scrape settings for kubeletScrapeInterval")
+    else
+      ConfigParseErrorLogger.logError(LOGGING_PREFIX, "kubeletScrapeInterval either not specified or not of type integer")
```

Comment:
I think this would be the place where you can add the int validation, since the value you are accepting is a string, you will need to parse it make sure the prefix is an integer and the suffix is a valid interval suffix.

Reply:
I fixed this now in the last commit.